pkg/prometheus/statefulset_test.go

```diff
@@ -1353,7 +1353,7 @@ func TestRetentionAndRetentionSize(t *testing.T) {
 		{"v2.5.0", "1d", "512MB", "--storage.tsdb.retention=1d", "--storage.tsdb.retention.size=512MB", true, false},
 		{"v2.7.0", "", "", "--storage.tsdb.retention.time=24h", "--storage.tsdb.retention.size=", true, false},
 		{"v2.7.0", "1d", "", "--storage.tsdb.retention.time=1d", "--storage.tsdb.retention.size=", true, false},
-		{"v2.7.0", "", "512MB", "--storage.tsdb.retention.time=24h", "--storage.tsdb.retention.size=512MB", false, true},
+		{"v2.7.0", "", "512MB", "--storage.tsdb.retention.time=24h", "--storage.tsdb.retention.size=512MB", true, true},
```

Comment:
can you elaborate on why the test needs a change?

Reply:
Maybe I misunderstood but in this test we were validating that if when`retention.time` is not set in the CR then the operator should not set `retention.time`. However from the code what I understood was that we should set `--storage.tsdb.retention.time=24h` if nothing was defined in the CR. So either I understood that wrongfully or the test needs to be fixed.

Reply:
IIRC the semantics are:
* if retentionSize and retention are empty then the retention argument is set to 24h.
* if retention is empty but not retentionSize, the retention argument isn't set (e.g. the only limit is space).
* if retentionSize is empty but not retention, the retention argument is set (e.g. the only limit is time).
* if both aren't empty, both arguments are set.

And if the Prometheus version is < v2.7.0 then the operator assumes that retentionSize is empty.

Now the test itself isn't 100% correct: if we expect that no retention CLI argument is defined, it should ensure that no argument with the said prefix (e.g.  `--storage.tsdb.retention.time`) exists (right now it verifies that `--storage.tsdb.retention.time=24h` isn't present).

Reply:
Should be good now, reverted my change to this part of the test